deepmd/pt/utils/env_mat_stat.py

```diff
@@ -101,6 +101,14 @@ def iter(
             dtype=env.GLOBAL_PT_FLOAT_PRECISION,
             device=env.DEVICE,
         )
+        if self.last_dim == 4:
+            radial_only = False
+        elif self.last_dim == 1:
+            radial_only = True
+        else:
+            raise ValueError(
+                "last_dim should be 1 for raial-only or 4 for full descriptor."
+            )
```

Comment:
assuming all the data share the same type of descriptor, otherwise, we need to move this into the for loop.